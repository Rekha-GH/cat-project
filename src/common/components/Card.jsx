import React from "react";
import HeartButton from "./HeartButton";
import TransparentButton from "./TransparentButton";

const DEFAULT_CLASSNAME = "card";
const IMAGE_CLASSNAME = `${DEFAULT_CLASSNAME}__image`;
const HEART_CLASSNAME = `${DEFAULT_CLASSNAME}__heart-icon`;
const IMAGE_DETAIL_CLASSNAME = `${DEFAULT_CLASSNAME}__image-detail`;

const Card = React.memo((props) => {
  return (
    <section className={DEFAULT_CLASSNAME}>
      <HeartButton
        classes={HEART_CLASSNAME}
        action={props.likeClicked}
        liked={props.liked}
      />
      <img className={IMAGE_CLASSNAME} alt={"cat"} src={props.imageUrl} />
      <div className={IMAGE_DETAIL_CLASSNAME}>
        <TransparentButton action={props.clickedUpVote}>👍</TransparentButton>
        <TransparentButton action={props.clickedDownVote}>👎</TransparentButton>
        {props.totalVote}
      </div>
    </section>
  );
});

export default Card;
