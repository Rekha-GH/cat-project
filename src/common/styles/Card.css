.card {
  overflow: hidden;
  position: relative;
  background: rgba(0, 0, 0, 0.7);
  margin: 0 auto;
  color: #fff;
}

.card > .card__image {
  width: 340px;
  height: 340px;
  object-fit: cover;
}

.card > .card__heart-icon {
  position: absolute;
  right: 0;
  background: #000;
}

.card > .card__image-detail {
  padding-bottom: 5px;
}
