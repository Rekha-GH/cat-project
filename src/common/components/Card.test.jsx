import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import * as ReactRedux from "react-redux";
import Card from "./Card";

Enzyme.configure({ adapter: new Adapter() });

describe("Card Test", () => {
  test("Card Call Correctly", () => {
    const example = {
      id: 1,
      url: "img.jpg",
      likeId: 2845857,
      totalVote: 2,
      voteIds: [],
    };

    const wrapper = shallow(<Card {...example} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
