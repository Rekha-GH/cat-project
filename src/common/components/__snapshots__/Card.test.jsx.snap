// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Test Card Call Correctly 1`] = `
<section
  className="card"
>
  <Memo()
    classes="card__heart-icon"
  />
  <img
    alt="cat"
    className="card__image"
  />
  <div
    className="card__image-detail"
  >
    <Memo()>
      👍
    </Memo()>
    <Memo()>
      👎
    </Memo()>
    2
  </div>
</section>
`;
